src/Servers/IIS/ResolveIisReferences.targets

```diff
@@ -8,23 +8,6 @@ with the right MSBuild incantations to get output copied to the right place.
   <Choose>
     <!-- IIS native projects only build on Windows with MSBuild.exe -->
     <When Condition="'$(OS)' == 'Windows_NT' AND '$(MSBuildRuntimeType)' != 'Core' ">
-      <ItemGroup Condition=" '@(Reference->AnyHaveMetadataValue('Identity', 'AspNetCoreModule'))' == 'true' ">
```

Comment:
cc @natemcmaster with regards to making this work.

Reply:
Making what work?

Reply:
I have never seen `<Reference Include="AspNetCoreModule">` work correctly (it fails on restore). I'm fine with removing it, but it would be nice to have a shortcut for referencing native projects.

Reply:
Problem with the shortcut is that it's almost never correct. You never need both handler and shim references from one project.

Reply:
I'm thinking of the following:
```msbuild
<ItemGroup Condition=" '@(Reference->AnyHaveMetadataValue('Identity', 'AspNetCoreModule'))' == 'true' ">
        <Reference Remove="AspNetCoreModuleV2" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModule\AspNetCore\AspNetCore.vcxproj"  Platform="x64" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModule\AspNetCore\AspNetCore.vcxproj" Platform="x86" />
</ItemGroup>

<ItemGroup Condition=" '@(Reference->AnyHaveMetadataValue('Identity', 'AspNetCoreModuleV2'))' == 'true' ">
        <Reference Remove="AspNetCoreModuleV2" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\AspNetCore\AspNetCore.vcxproj"  Platform="x64" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\OutOfProcessRequestHandler\OutOfProcessRequestHandler.vcxproj" HandlerPath="2.0.0"  Platform="x64" />

        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\AspNetCore\AspNetCore.vcxproj" Platform="x86" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\OutOfProcessRequestHandler\OutOfProcessRequestHandler.vcxproj" HandlerPath="2.0.0" Platform="x86" />
</ItemGroup>

<ItemGroup Condition=" '@(Reference->AnyHaveMetadataValue('Identity', 'AspNetCoreModuleInProcess'))' == 'true' ">
        <Reference Remove="AspNetCoreModuleInProcess" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\InProcessRequestHandler\InProcessRequestHandler.vcxproj"  Platform="x64" />
        <NativeProjectReference Include="$(MSBuildThisFileDirectory)AspNetCoreModuleV2\InProcessRequestHandler\InProcessRequestHandler.vcxproj"  Platform="x86" />
</ItemGroup>
```

Reply:
Why are we removing this? Was isn't working correctly?

Reply:
Because people don't understand where to use it (almost nowhere) and I don't think there is still need for this magic.

Reply:
Which people? The goal of this code was to simplify what it takes to reference native projects. I'm not sure how NativeProjectReference is any less magical.

Reply:
Talked in person. Let's make NativeProjectReference part of eng/targets/ResolveReferences.targets and document it in docs/ReferenceResolution.md

Reply:
I ended up not needing to use `NativeProjectReference` in WIX projects